helm/cluster/templates/clusterapi/controlplane/_helpers_clusterconfiguration_apiserver.tpl

```diff
@@ -25,8 +25,8 @@ extraArgs:
   cloud-provider: external
   enable-admission-plugins: {{ include "cluster.internal.controlPlane.kubeadm.clusterConfiguration.apiServer.enableAdmissionPlugins" $ }}
   encryption-provider-config: /etc/kubernetes/encryption/config.yaml
-  {{- if $.Values.providerIntegration.controlPlane.kubeadmConfig.clusterConfiguration.apiServer.etcdPrefix }}
-  etcd-prefix: {{ $.Values.providerIntegration.controlPlane.kubeadmConfig.clusterConfiguration.apiServer.etcdPrefix }}
+  {{- if $.Values.internal.advancedConfiguration.controlPlane.apiServer.etcdPrefix }}
+  etcd-prefix: {{ $.Values.internal.advancedConfiguration.controlPlane.apiServer.etcdPrefix }}
```

Comment:
If we anyway changing the contract a bit, WDYT about moving the etcdPrefix into etcd?

It's a flag on the API server but at the same time it's ETCD configuration. 
Maybe it's better to have all ETCD configurations under one roof (etcd)

Reply:
IIUC it's actual API server config, and how API server uses etcd as a client/consumer (`--etcd-prefix` flag is passed to API server, not to etcd, you can find it [here](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/) in API server CLI flags docs), i.e. it's about how the API server will name the resources it creates for itself.

Some other other etcd client would use another prefix, so different clusters can use different prefixes and then share the same etcd cluster (e.g. see this example issue [here](https://github.com/kubernetes/kubeadm/issues/1503)).

The name of flag `--etcd-prefix` is not the best TBH 😄 it's a bit misleading.